inference-engine/src/gna_plugin/gna_graph_patterns.hpp

```diff
@@ -133,16 +133,17 @@ inline bool MustBeConvertedFromNCHWToNHWC(const std::vector<InferenceEngine::CNN
 }
 
 /**
- * @brief returns rotation information for a layer based on the previous convolution or pooling dimensions order
- * @param layer layer from which rotation info search must be started
- * @return bool value which identifies if rotation info is found and rotation information
+ * @brief returns transposition information for a layer based on the previous convolution or pooling dimensions order
+ * @param layer layer from which transposition info search must be started
+ * @return bool value which identifies if transposition info is found and transposition information
  */
 inline std::vector<TranspositionInfo> FindTranspositionInfoFromPrevLayers(InferenceEngine::CNNLayerPtr layer) {
     std::function<std::vector<TranspositionInfo>(InferenceEngine::CNNLayerPtr)> findTranspositionInfoRecursive =
         [&findTranspositionInfoRecursive](InferenceEngine::CNNLayerPtr layer) -> std::vector<TranspositionInfo> {
         if (LayerInfo(layer).isConvolution() || LayerInfo(layer).isPooling()) {
-            auto out_dims = layer->outData[0]->getDims();
-            return {{true, out_dims[1], out_dims[2] * out_dims[3]}};
+            auto rows = FROM_IR_DIM(layer->outData[0], 3);
+            auto columns = FROM_IR_DIM(layer->outData[0], 1) * FROM_IR_DIM(layer->outData[0], 2);
+            return {{true, rows, columns}};
```

Comment:
It's not quite the same. If output layout is CHW instead of NCHW out_dims[1] will return incorrect result (H instead of C).   But FROM_IR_DIM(layer->outData[0], 3) will return correct C dimension since it calculates it using an offset from the last dimension. At the same time I implemented more readable format of function in another PR, it can be reused here after that PR will be merged.

Reply:
OK